tests/integration/run_modes/test_mode_overrides.py

```diff
@@ -0,0 +1,169 @@
+async def test_run_mode_override_from_config(
+    capture_live_submissions,
+    flow,
+    scheduler,
+    run,
+    complete,
+    workflow_run_mode,
+    validate
+):
+    """Test that `[runtime][<namespace>]run mode` overrides workflow modes."""
+    id_ = flow({
+        'scheduling': {
+            'graph': {
+                'R1': 'live & skip',
+            },
+        },
+        'runtime': {
+            'root': {'simulation': {'default run length': 'PT0S'}},
+            'live': {'run mode': 'live'},
+            'skip': {'run mode': 'skip'},
+        }
+    })
+    run_mode = RunMode(workflow_run_mode)
+    validate(id_)
+    schd = scheduler(id_, run_mode=run_mode, paused_start=False)
+    async with run(schd):
+        await complete(schd)
+
+    if workflow_run_mode == 'live':
+        assert capture_live_submissions() == {'1/live'}
+    elif workflow_run_mode == 'dummy':
+        # Skip mode doesn't override dummy mode:
+        assert capture_live_submissions() == {'1/live', '1/skip'}
+    else:
+        assert capture_live_submissions() == set()
+
+
+async def test_force_trigger_does_not_override_run_mode(
+    flow,
+    scheduler,
+    start,
+):
+    """Force-triggering a task will not override the run mode.
+
+    Taken from spec at
+    https://github.com/cylc/cylc-admin/blob/master/docs/proposal-skip-mode.md#proposal
+    """
+    wid = flow({
+        'scheduling': {'graph': {'R1': 'foo'}},
+        'runtime': {'foo': {'run mode': 'skip'}}
+    })
+    schd = scheduler(wid, run_mode="live")
+    async with start(schd):
+        foo = schd.pool.get_tasks()[0]
+
+        # Force trigger task:
+        schd.pool.force_trigger_tasks('1/foo', [1])
+
+        # ... but job submission will always change this to the correct mode:
+        schd.task_job_mgr.submit_task_jobs(
+            schd.workflow,
+            [foo],
+            schd.server.curve_auth,
+            schd.server.client_pub_key_dir)
+
+        assert foo.run_mode.value == 'skip'
+
+
+async def test_run_mode_skip_abides_by_held(
+    flow,
+    scheduler,
+    run,
+    complete
+):
+    """Tasks with run mode = skip will continue to abide by the
+    is_held flag as normal.
+
+    Taken from spec at
+    https://github.com/cylc/cylc-admin/blob/master/docs/proposal-skip-mode.md#proposal
+    """
+    wid = flow({
+        'scheduling': {'graph': {'R1': 'foo'}},
+        'runtime': {'foo': {'run mode': 'skip'}}
+    })
+    schd = scheduler(wid, run_mode="live", paused_start=False)
+    async with run(schd):
+        foo = schd.pool.get_tasks()[0]
+        assert foo.state.is_held is False
+
+        # Hold task, check that it's held:
+        schd.pool.hold_tasks('1/foo')
+        assert foo.state.is_held is True
+
+        # Run to completion, should happen if task isn't held:
+        with pytest.raises(
+            Exception,
+            match="Timeout waiting for workflow to shut down"
+        ):
+            await complete(schd, timeout=5)
```

Comment:
I get what you're trying to do, but in a sense this test isn't valid, as held tasks do not get queued or released from queues, which comes before job submission. So it's kind of a waste of 5 seconds.

Reply:
> as held tasks do not get queued or released from queues

That's an explanation for _why_ skip mode tasks will behave like any other when held. Understanding the mechanism (there's nothing I've done which _should_ cause skip mode tasks to ignore holds) doesn't mean that I shouldn't test it. 

Admittedly this is more of a functional test. Would you be objecting if I'd written this using the bash framework?

Reply:
https://github.com/wxtim/cylc/pull/67